plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -0,0 +1,520 @@
+                });
+
+        IcebergTableOperations operations = tableOperationsProvider.createTableOperations(
+                this,
+                session,
+                table.getSchemaName(),
+                table.getTableName(),
+                Optional.empty(),
+                Optional.empty());
+        operations.initializeFromMetadata(metadata);
+        return new BaseTable(operations, quotedTableName(table));
+    }
+
+    @Override
+    public boolean dropTable(ConnectorSession session, SchemaTableName schemaTableName, boolean purgeData)
+    {
+        Table table = loadTable(session, schemaTableName);
+        validateTableCanBeDropped(table, schemaTableName);
+        try {
+            deleteTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+        }
+        catch (AmazonServiceException e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, e);
+        }
+
+        Path tableLocation = new Path(table.location());
+        if (purgeData) {
+            try {
+                hdfsEnvironment.getFileSystem(new HdfsEnvironment.HdfsContext(session), tableLocation).delete(tableLocation, true);
+            }
+            catch (Exception e) {
+                // don't fail if unable to delete path
+                log.warn(e, "Failed to delete path: " + tableLocation);
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public Transaction newCreateTableTransaction(
+            ConnectorSession session,
+            SchemaTableName schemaTableName,
+            Schema schema,
+            PartitionSpec partitionSpec,
+            String location,
+            Map<String, String> properties)
+    {
+        TableMetadata metadata = newTableMetadata(schema, partitionSpec, location, properties);
+        TableOperations tableOperations = tableOperationsProvider.createTableOperations(
+                this,
+                session,
+                schemaTableName.getSchemaName(),
+                schemaTableName.getTableName(),
+                Optional.of(session.getUser()),
+                Optional.of(location));
+        return createTableTransaction(schemaTableName.toString(), tableOperations, metadata);
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to)
+    {
+        boolean newTableCreated = false;
+        boolean oldTableDeleted = false;
+        try {
+            GetTableRequest getTableRequest = new GetTableRequest()
+                    .withDatabaseName(from.getSchemaName())
+                    .withName(from.getTableName());
+            catalogId.ifPresent(getTableRequest::setCatalogId);
+            com.amazonaws.services.glue.model.Table table = stats.getGetTable().call(() -> glueClient.getTable(getTableRequest).getTable());
+            TableInput tableInput = new TableInput()
+                    .withName(to.getTableName())
```

Comment:
`io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog#newCreateTableTransaction` also creates table copy, ie a new TableInput from `com.amazonaws.services.glue.model.Table`
yet, the logic differs heavily, for no apparent reason

have a method that unifies the logic, as it should stay in sync.